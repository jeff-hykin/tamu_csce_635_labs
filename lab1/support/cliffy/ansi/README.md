# Ansi

<p align="center" class="badges-container">
  <a href="https://github.com/c4spar/deno-cliffy/releases">
    <img alt="Version" src="https://img.shields.io/github/v/release/c4spar/deno-cliffy?logo=github&color=blue" />
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/actions/workflows/test.yml">
    <img alt="Build status" src="https://github.com/c4spar/deno-cliffy/workflows/Test/badge.svg?branch=main" />
  </a>
  <a href="https://codecov.io/gh/c4spar/deno-cliffy">
    <img src="https://codecov.io/gh/c4spar/deno-cliffy/branch/main/graph/badge.svg"/>
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/labels/module%3Aansi-escape">
    <img alt="issues" src="https://img.shields.io/github/issues/c4spar/deno-cliffy/module:ansi-escape?label=issues&logo=github&color=yellow">
  </a>
  <a href="https://deno.land/">
    <img alt="Deno version" src="https://img.shields.io/badge/deno-^1.4.0-blue?logo=deno" />
  </a>
  <a href="https://doc.deno.land/https/deno.land/x/cliffy/ansi/mod.ts">
    <img alt="doc" src="https://img.shields.io/badge/deno-doc-yellow?logo=deno" />
  </a>
  <a href="https://discord.gg/ghFYyP53jb">
    <img alt="Discord" src="https://img.shields.io/badge/join-chat-blue?logo=discord&logoColor=white" />
  </a>
  <a href="../LICENSE">
    <img alt="License" src="https://img.shields.io/github/license/c4spar/deno-cliffy?logo=github" />
  </a>
  <br>
  <a href="https://deno.land/x/cliffy">
    <img alt="Discord" src="https://img.shields.io/badge/Published on deno.land-blue?logo=deno&logoColor=959DA6&color=272727" />
  </a>
</p>

The documentation has been moved to https://cliffy.io/docs/ansi.
