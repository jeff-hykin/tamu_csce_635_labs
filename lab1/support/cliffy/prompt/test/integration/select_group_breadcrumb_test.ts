import { ansi } from "../../../ansi/ansi.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";
import { Select } from "../../select.ts";

await snapshotTest({
  name: "select prompt with groups and breadcrumb",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should render breadcrumb": {
      stdin: ansi
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .toArray(),
    },
  },
  async fn() {
    await Select.prompt({
      message: "Select an option",
      maxBreadcrumbItems: 4,
      options: [{
        name: "Foo",
        options: [{
          name: "Bar",
          options: [{
            name: "Baz",
            options: [{
              name: "Beep",
              options: [{
                name: "Boop",
                options: [{
                  name: "Fiz",
                  options: [{
                    name: "Faz",
                    value: "123",
                  }],
                }],
              }],
            }],
          }],
        }],
      }],
    });
  },
});

await snapshotTest({
  name: "select prompt with custom breadcrumb separator",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should render breadcrumb": {
      stdin: ansi
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .text("\n")
        .toArray(),
    },
  },
  async fn() {
    await Select.prompt({
      message: "Select an option",
      breadcrumbSeparator: " # ",
      maxBreadcrumbItems: 4,
      options: [{
        name: "Foo",
        options: [{
          name: "Bar",
          options: [{
            name: "Baz",
            options: [{
              name: "Beep",
              options: [{
                name: "Boop",
                options: [{
                  name: "Fiz",
                  options: [{
                    name: "Faz",
                    value: "123",
                  }],
                }],
              }],
            }],
          }],
        }],
      }],
    });
  },
});
