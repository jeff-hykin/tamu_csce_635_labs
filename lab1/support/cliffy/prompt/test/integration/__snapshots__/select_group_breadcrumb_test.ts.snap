export const snapshot = {};

snapshot[`select prompt with groups and breadcrumb > should render breadcrumb 1`] = `
stdout:
"? Select an option
❯ 📁 Foo\\x1b[1A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo
 ❯ 📁 Bar\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo › Bar
 ❯ 📁 Baz\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo › Bar › Baz
 ❯ 📁 Beep\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo › Bar › Baz › Beep
 ❯ 📁 Boop\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo › .. › Baz › Beep › Boop
 ❯ 📁 Fiz\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo › .. › Beep › Boop › Fiz
 ❯ Faz\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option › Faz
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`select prompt with custom breadcrumb separator > should render breadcrumb 1`] = `
stdout:
"? Select an option
❯ 📁 Foo\\x1b[1A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo
 ❯ 📁 Bar\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo # Bar
 ❯ 📁 Baz\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo # Bar # Baz
 ❯ 📁 Beep\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo # Bar # Baz # Beep
 ❯ 📁 Boop\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo # .. # Baz # Beep # Boop
 ❯ 📁 Fiz\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  📂 Foo # .. # Beep # Boop # Fiz
 ❯ Faz\\x1b[2A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option › Faz
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
